src/main/java/org/wildfly/security/auth/provider/JaasSecurityRealm.java

```diff
@@ -100,13 +103,47 @@ public CredentialSupport getCredentialSupport(Class<?> credentialType) throws Re
         }
     }
 
-    private class JAASRealmIdentity implements RealmIdentity {
+    private LoginContext createLoginContext(final String loginConfig, final Subject subject, final CallbackHandler handler) throws RealmUnavailableException {
+        if (WildFlySecurityManager.isChecking()) {
+            try {
+                return AccessController.doPrivileged(new CreateLoginContextAction(loginConfig, subject, handler));
```

Comment:
At some point I would like to change this, it is not critical for this pull request but I prefer a pattern where the code within the privileged action is also used for the non-privileged case.

I did find a bug a while back where code that is split like this had diverged so you ended up with a bug only once the security manager was enabled.